app/src/main/java/ch/epfl/skysync/screens/reports/CrewReport.kt

```diff
@@ -20,123 +21,141 @@ import androidx.compose.ui.Modifier
+                value = prestigeChampagne,
+                onValueChange = { value -> prestigeChampagne = value.filter { it.isDigit() } },
+                padding = defaultPadding,
+                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                isError = errorPrestigeChampagne,
+                messageError = if (errorPrestigeChampagne) "Please enter a number" else "")
+          }
 
-      Button(
-          modifier = Modifier.fillMaxWidth().padding(defaultPadding).testTag("Submit Button"),
-          colors = ButtonDefaults.buttonColors(containerColor = lightOrange),
-          onClick = {
-            errorLittleChampagne = !bottleInputValidation(littleChampagne)
-            errorBigChampagne = !bottleInputValidation(bigChampagne)
-            errorPrestigeChampagne = !bottleInputValidation(prestigeChampagne)
-            if (!hasError(errorLittleChampagne, errorBigChampagne, errorPrestigeChampagne)) {
-              CrewReport(
-                  author = crew.id,
-                  begin = beginTime,
-                  end = endTime,
-                  pauseDuration = pauseDuration.toInt(),
-                  comments = comments,
-              )
-              // TODO save report
-              navHostController.navigate(Route.CREW_HOME) {
-                popUpTo(Route.CREW_HOME) { inclusive = true }
+          baseReportFields(
+              defaultPadding = defaultPadding,
+              beginTime = beginTime,
+              endTime = endTime,
+              pauseDuration = pauseDuration,
+              flight = flight.value!!,
+              onConfirm = { vehicle, problem ->
+                newVehicle = vehicle
+                newProblem = problem
+                addProblem = true
+              },
+              vehicleProblem = vehicleProblem,
+              comments = comments,
+              onBeginTimeChange = { beginTime = it },
+              onEndTimeChange = { endTime = it },
+              onPauseDurationChange = { pauseDuration = it },
+              onCommentsChange = { comments = it })
+        }
+
+        Button(
+            modifier = Modifier.fillMaxWidth().padding(defaultPadding).testTag("Submit Button"),
+            colors = ButtonDefaults.buttonColors(containerColor = lightOrange),
+            onClick = {
+              errorLittleChampagne = !bottleInputValidation(littleChampagne)
+              errorBigChampagne = !bottleInputValidation(bigChampagne)
+              errorPrestigeChampagne = !bottleInputValidation(prestigeChampagne)
+              if (!hasError(errorLittleChampagne, errorBigChampagne, errorPrestigeChampagne)) {
+                CrewReport(
+                    author = crew.value!!.id,
+                    begin = beginTime,
+                    end = endTime,
+                    pauseDuration = pauseDuration.toInt(),
+                    comments = comments,
+                )
+                // TODO save report
```

Comment:
Should `finishedFlightsViewModel.addReport` be called here ?

Reply:
Yes, you're right